pybot/endpoints/slack/utils/event_utils.py

```diff
@@ -103,3 +106,69 @@ def build_messages(user_id) -> Tuple[Message, Message, Message, Message]:
         data = await response.json()
         headers = {"Authorization": f"Bearer {data['token']}"}
     return headers
+
+
+async def get_profile_suggestions(
+    slack_id: int,
+    auth_header: Dict[str, str],
+    slack_api: SlackAPI,
+    session: ClientSession,
+) -> List[Tuple[str, str]]:
+    """
+    Generates a list of slack channels to suggest
+
+    :param slack_id: The slack_id to return profile information from
+    :return: A list of slack channel names to suggest for the user
+    """
+    results = []
+
+    # Get profile data that we want to build suggestions from.
+    user_info = await slack_api.query(methods.USERS_INFO, {"user": slack_id})
+    email = user_info["user"]["profile"]["email"]
+
+    async with session.get(
+        f"{BACKEND_URL}/auth/profile/admin/",
+        headers=auth_header,
+        params={"email": email},
+    ) as response:
+        data = await response.json()
+        logger.info(f"Retrieved profile data from email: {email}")
+
+    # Get the names of all public, non-archived channels
+    channels = await slack_api.query(
+        methods.CONVERSATION_LIST, {"exclude_archived": True}
+    )
+
+    # Get a list of all the profile data we want to compare
+    possible_suggestions = parse_suggestions_from_profile(data)
+
+    # For each value in the profile data, check that we have a channel name that matches.
+    for info in possible_suggestions:
+        for channel in channels:
+            if info in channel[1]:
```

Comment:
Not sure how you got that the channel can be indexed...

```json 
"channels": [
        {
            "id": "C0G9QF9GW",
            "name": "random",
            "is_channel": true,
            "created": 1449709280,
            "creator": "U0G9QF9C6",
            "is_archived": false,
            "is_general": false,
            "name_normalized": "random",
            "is_shared": false,
            "is_org_shared": false,
            "is_member": true,
            "is_private": false,
             ........
        },
```

I think you need to compare the channel['name'] to the suggestion. But you should also exclude `channel['is_private']` as we don't want to make the names of these channels visible.

Reply:
Oh thanks for catching that. 👍

Reply:
Based on the above, ignore the `is_private` part but we still need to make sure this works.